site/app/controllers/admin/ReportController.php

```diff
@@ -21,32 +22,20 @@
 /**
  * Class ReportController
  * @package app\controllers\admin
- *
+ * @AccessControl(role="INSTRUCTOR")
  */
 class ReportController extends AbstractController {
+    /**
+     * @deprecated
+     */
     public function run() {
-        switch ($_REQUEST['action']) {
-            case 'csv':
-                $this->generateCSVReport();
-                break;
-            case 'summary':
-                $this->generateGradeSummaries();
-                break;
-            case 'reportpage':
-            default:
-                $this->showReportPage();
-                break;
-        }
+        return null;
     }
```

Comment:
Is there a reason to leave this function in at all given that it's just returning null?

Reply:
AbstractController enforces that. Needs to be removed after old-style routes completely phase out.

Reply:
Oh, yeah, that makes sense.